aws-api/src/main/java/com/amplifyframework/api/aws/AWSApiPlugin.java

```diff
@@ -130,44 +133,77 @@ public void configure(@NonNull JSONObject pluginConfigurationJson, Context conte
         return Collections.unmodifiableMap(apiClientsByName);
     }
 
+    @Nullable
     @Override
     public <T extends Model> GraphQLOperation<T> query(
             @NonNull Class<T> modelClass,
             @NonNull ResultListener<GraphQLResponse<Iterable<T>>> responseListener
     ) {
-        final String apiName = getSelectedApiName(EndpointType.GRAPHQL);
+        final String apiName;
+        try {
+            apiName = getSelectedApiName(EndpointType.GRAPHQL);
+        } catch (ApiException exception) {
+            responseListener.onError(exception);
+
+            return null;
+        }
         return query(apiName, modelClass, responseListener);
     }
 
+    @Nullable
     @Override
     public <T extends Model> GraphQLOperation<T> query(
             @NonNull Class<T> modelClass,
             @NonNull String objectId,
             @NonNull ResultListener<GraphQLResponse<T>> responseListener
     ) {
-        final String apiName = getSelectedApiName(EndpointType.GRAPHQL);
+        final String apiName;
+        try {
+            apiName = getSelectedApiName(EndpointType.GRAPHQL);
+        } catch (ApiException exception) {
+            responseListener.onError(exception);
+
+            return null;
+        }
         return query(apiName, modelClass, objectId, responseListener);
     }
 
+    @Nullable
     @Override
     public <T extends Model> GraphQLOperation<T> query(
             @NonNull Class<T> modelClass,
             QueryPredicate predicate,
             @NonNull ResultListener<GraphQLResponse<Iterable<T>>> responseListener
     ) {
-        final String apiName = getSelectedApiName(EndpointType.GRAPHQL);
+        final String apiName;
+        try {
+            apiName = getSelectedApiName(EndpointType.GRAPHQL);
+        } catch (ApiException exception) {
+            responseListener.onError(exception);
+
+            return null;
+        }
         return query(apiName, modelClass, predicate, responseListener);
     }
 
+    @Nullable
     @Override
     public <T> GraphQLOperation<T> query(
             @NonNull GraphQLRequest<T> graphQLRequest,
             @NonNull ResultListener<GraphQLResponse<Iterable<T>>> responseListener
     ) {
-        final String apiName = getSelectedApiName(EndpointType.GRAPHQL);
+        final String apiName;
```

Comment:
I noticed this code snippet gets repeated in multiple methods. Would it be better to extract it to a private method?

Reply:
i thought about doing that, but something just didn't feel right about nesting private method the third time, not that there's anything conventionally wrong with it.

Reply:
also this code snippet is both setting the value of a string and returning null operation if an error is thrown, so the utility code wouldn't simplify it much